### PR TITLE
[CEED] Unify and Simplify the integration of PA Integrators using libCEED.

### PR DESCRIPTION
**Pull request by YohannDudouit**
This PR factorize the code that was common between `mass` and `diffusion` into a new function `CeedPAAssemble` and use this function to initialize the mass and diffusion Ceed Operators.

 | PR | Author | Editor | Reviewers | Assignment | Approval | Merge| 
 | --- | --- | --- | --- | --- | --- | --- | 
| [#1349](https://github.com/mfem/mfem/pull/1349) | @YohannDudouit | @v-dobrev | @camierjs + @barker29 | 03/12/20 | 03/23/20 | 03/26/20 | |

### CODE REVIEW

**Comment by v-dobrev** [comment]
It seems like this needs `make style`.

**Comment by camierjs** [comment]
We have a few of unused variables:
```
fem/libceed/ceed.cpp:157:33: warning: unused variable 'mesh_fes' [-Wunused-variable]
      const FiniteElementSpace *mesh_fes = mesh->GetNodalFESpace();
                                ^
fem/libceed/ceed.cpp:91:14: warning: unused variable 'order' [-Wunused-variable]
   const int order = fes.GetOrder(0);
             ^
fem/libceed/ceed.cpp:289:14: warning: unused variable 'ir_order' [-Wunused-variable]
   const int ir_order = irm.GetOrder();
```

**Comment by YohannDudouit** [comment]
@camierjs I addressed your issues I think.

File: fem/libceed/ceed.hpp [hunk 1]
```diff
@@ -105,6 +118,11 @@ void InitCeedBasisAndRestriction(const FiniteElementSpace &fes,
 /// Return the path to the libCEED q-function headers.
 const std::string &GetCeedPath();
 
+void CeedPAAssemble(const FiniteElementSpace &fes,
```
Comment (barker29) [inline fem/libceed/ceed.hpp]: Can we get a couple lines of documentation here?

Comment (YohannDudouit) [reply fem/libceed/ceed.hpp]: Done, tell me if you like it or if you think I should document more specific points.

**Comment by tzanio** [comment]
LGTM

**Comment by tzanio** [comment]
Merged in `next` for testing...

**Comment by tzanio** [comment]
Are there any expected regression testing differences due to this PR?

There are a lot diffs in tonight's autotest run on  [tux375](https://mybitbucket.llnl.gov/projects/MFEM/repos/autotest/commits/8d65dbd761991b80662a1f2b8e7126690e05f7c7#tux375/2020-03-24-llnl-gh-next/sample-runs-baseline-tux375.diff) (you need to clone it locally to view) and this seems the most likely cause for them... Can you please investigate?

File: fem/libceed/ceed.hpp [hunk 2]
```diff
@@ -91,17 +91,38 @@ struct CeedData
 
 };
 
+/** This structure contains the data to assemble a PA operator with libCEED.
+ *  See libceed/mass.cpp or libceed/diffusion.cpp for examples. */
```
Comment (barker29) [inline fem/libceed/ceed.hpp]: (minor) The `*` at the beginning of the line in the comment block is not consistent with the rest of the Doxygen comments in MFEM (also appears several other places in this file).

Comment (tzanio) [reply fem/libceed/ceed.hpp]: I think I "fixed" this before merging... did I miss an instance?

Comment (barker29) [reply fem/libceed/ceed.hpp]: No, I just followed some github notification blindly to an outdated diff. Everything is fine, never mind.

Comment (tzanio) [reply fem/libceed/ceed.hpp]: Ok. I agree with your point though -- we should try to unify the style :)

Comment (YohannDudouit) [reply fem/libceed/ceed.hpp]: Sorry vscode is adding these automatically... I'll be more careful.

**Pull request merged**